bridge/opencensus/aggregation.go

```diff
@@ -0,0 +1,174 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opencensus
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"go.opencensus.io/metric/metricdata"
+
+	"go.opentelemetry.io/otel/metric/number"
+	"go.opentelemetry.io/otel/sdk/export/metric/aggregation"
+)
+
+var (
+	errIncompatibleType = errors.New("incompatible type for aggregation")
+	errEmpty            = errors.New("points may not be empty")
+	errBadPoint         = errors.New("point cannot be converted")
+)
+
+// aggregationWithEndTime is an aggregation that can also provide the timestamp
+// of the last recorded point.
+type aggregationWithEndTime interface {
+	aggregation.Aggregation
+	end() time.Time
+}
+
+// newAggregationFromPoints creates an OpenTelemetry aggregation from
+// OpenCensus points.  Points may not be empty.
+func newAggregationFromPoints(points []metricdata.Point) (aggregationWithEndTime, error) {
+	if len(points) == 0 {
+		return nil, errEmpty
+	}
+	switch t := points[0].Value.(type) {
+	case int64:
+		return newExactAggregator(points)
+	case float64:
+		return newExactAggregator(points)
+	case *metricdata.Distribution:
+		return newDistributionAggregator(points)
+	default:
+		// TODO add *metricdata.Summary support
+		return nil, fmt.Errorf("%w: %v", errIncompatibleType, t)
+	}
+}
+
+var _ aggregation.Aggregation = &ocExactAggregator{}
+var _ aggregation.LastValue = &ocExactAggregator{}
+var _ aggregation.Points = &ocExactAggregator{}
+
+// newExactAggregator creates an OpenTelemetry aggreation from OpenCensus points.
+// Points may not be empty, and must only contain integers or floats.
+func newExactAggregator(pts []metricdata.Point) (aggregationWithEndTime, error) {
+	points := make([]number.Number, len(pts))
+	for i, pt := range pts {
+		switch t := pt.Value.(type) {
+		case int64:
+			points[i] = number.NewInt64Number(pt.Value.(int64))
+		case float64:
+			points[i] = number.NewFloat64Number(pt.Value.(float64))
+		default:
+			return nil, fmt.Errorf("%w: %v", errIncompatibleType, t)
+		}
+	}
+	return &ocExactAggregator{
+		points:  points,
+		endTime: pts[len(pts)-1].Time,
```

Comment:
Are all values in `points` guaranteed to have the same ending time or be monotonically increasing with respect to time?  Should this assumption be documented?

Reply:
Looking at where the list of points is constructed: [raw metric src](https://github.com/census-instrumentation/opencensus-go/blob/master/metric/common.go#L86), [view src](https://github.com/census-instrumentation/opencensus-go/blob/master/stats/view/view_to_metric.go#L124), there is only ever a single point in the list of points.  I'll document it.